docs/manual/writing-modules/types.md

```diff
@@ -0,0 +1,368 @@
+# Option Types {#sec-option-types}
+
+Overall the basic option types are the same in Home Manager as NixOS. A
+few Home Manager options, however, make use of custom types that are
+worth describing in more detail. These are the option types `dagOf` and
+`gvariant` that are used, for example, by
+[programs.ssh.matchBlocks](#opt-programs.ssh.matchBlocks) and [dconf.settings](#opt-dconf.settings).
+
+[]{#sec-option-types-dag}`hm.types.dagOf`
```

Comment:
This seems to generate invalid HTML (my formatting):

``` html
<span class="term">
  <a id="sec-option-types-dag" />
  <code class="literal">hm.types.dagOf</code>
</span>
```

Firefox complains about this use of `<a>`, saying

> Self-closing /> is invalid syntax on non-void HTML elements.

The old manual contains

``` html
<span class="term">
  <a id="sec-option-types-dag"></a>
  <code class="literal">hm.types.dagOf</code>
</span>
```

and Firefox is happy with that 🙂

Doesn't matter much for this PR but may be worth updating the generator in Nixpkgs?

Reply:
I don't think I had to do anything special to make DocBook produce anchors of the form `<a id="blah"></a>` so I'm not sure why they would be different in Nixpkgs.

In any case, this messes up the rendering pretty badly.

This PR:
![swappy-20231130_000523](https://github.com/nix-community/home-manager/assets/798147/afbc0698-294b-491a-a044-557aa931b24d)

Current:
![swappy-20231130_000835](https://github.com/nix-community/home-manager/assets/798147/fd8a0f77-22fb-482f-b72c-38616cadc2d1)

Reply:
I don't understand why there's an empty anchor tag in the generated manual. This doesn't show in the rendered output. Is this for automated scraping of links?

Reply:
I can verify that the nixpkgs manual contains the same broken HTML output:
```
<a id="ssec-gnome-hooks-glib" /> <code class="literal">glib</code> 
```

Reply:
Even after changing the Content-Type to `application/xhtml+xml; charset=utf8` it renders like this:
![xhtml](https://github.com/nix-community/home-manager/assets/217918/292b290e-4318-4951-ae8f-bc3f0d8f30e1)

Reply:
For some reason `firefox` renders this correctly but not the `home-manager` manual. Moreover, fetching the `nixpkgs` manual via curl doesn't have this same issue (I built `nix build .#htmlDocs.nixpkgsManual` on the most recent `master` when producing the above HTML). The web page has
```html
<a id="ssec-gnome-hooks-glib" shape="rect"></a> <code class="literal">glib</code> 
```